subsys/modbus/modbus_serial.c

```diff
@@ -44,11 +44,32 @@ static void modbus_serial_tx_off(struct modbus_context *ctx)
 	struct modbus_serial_config *cfg = ctx->cfg;
 
 	uart_irq_tx_disable(cfg->dev);
+#ifndef CONFIG_MODBUS_SERIAL_USE_TXC
```

Comment:
All this is not acceptable because of:
- meaningless use of `ifdef` & co
- it is not necessary because problem is in the UART driver
- we can not have a driver whose behavior depends on the subsystem

Please close this PR or I will do it later.